---
#############################################
#  Active-Window Reconstruction Parameters  #
#############################################
reconstruction:
  active_window:
    find_visible_blocks_accurately: false
    max_frame_data_buffer_size: 300
    store_every_n_frame_data: 1
    min_object_allocation_confidence: 0.5
    min_object_volume: 0.0  # m^3
    max_object_volume: 8.0  # m^3
    only_extract_reconstructed_objects: true
    min_object_reconstruction_confidence: 0.5
    scale_object_reconstruction_confidence_with_weight: true
    min_object_reconstruction_observations: 3
    object_reconstruction_resolution: -0.05  # Positive: voxel size in meters. Negative: fraction of the extent. 0: Skip. (-0.02)
    object_detector:
      min_cluster_size: 20  # pixels
      max_cluster_size: 50000  # pixels
      use_full_connectivity: true
      use_3d: true
      grid_size: 0.1  # m
    tracker:
      semantic_association: assign_track # 'assign_cluster', 'assign_track'
      min_semantic_iou: 0.6
      min_cosine_sim: 0.7
      min_cross_iou: 0.1
      temporal_window: $(arg temporal_window_size_s)
      min_num_observations: 5
#########################
#  Frontend Parameters  #
#########################
frontend:
  view_database:
    selection_method: boundary
########################
#  Backend Parameters  #
########################
backend:
  objects:
    min_segment_score: 0.0
    selector:
      max_delta: 1.0e-5
      score_threshold: 0.23
      top_k: 2
      cumulative: true
      null_task_preprune: true